### PR TITLE
JavaScript cleanup

### PR DESCRIPTION
**Pull request by mxstbr**
Note: completely untested code 😁 you should definitely verify this still works as expected before merging.

I will add comments inline to explain the changes I made!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,15 +1,13 @@
 // Set up auth for Google spreadsheet
-var GoogleSpreadsheet = require('google-spreadsheet');
-var creds = require('./google-sheets-token.json');
-const { promisify } = require('util');
+const GoogleSpreadsheet = require('google-spreadsheet')
+const creds = require('./google-sheets-token.json')
```
Comment (mxstbr) [inline index.js]: In modern JS, prefer using `const` and `let` over `var`. The difference between `const` and `let` is that variables defined as `const`s cannot be changed, while `let`s can be. 

Using the exclusively increases code clarity, as I can read the code and discern which variables are going to change (which in general should be avoided as much as possible) vs. which ones are going to stay constant, making the code easier to follow.

File: index.js [hunk 2]
```diff
@@ -26,13 +24,13 @@ module.exports = app => {
   app.on('issue_comment.edited', async context => {
     const regex = /(\-\s\[x\]\s.(.*?)\:)/i
     app.log(context.payload.comment.body)
-    var comment_body = context.payload.comment.body
-    var url = context.payload.issue.html_url
-    var key = ""
+    const comment_body = context.payload.comment.body
+    const url = context.payload.issue.html_url
+    let key = ""
```
Comment (mxstbr) [inline index.js]: See this? Before I had no idea which variables would be mutated and which ones would stay the same. Now I can read the code and immediately know! 👍

File: index.js [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
 
 // Create a document object using the ID of the spreadsheet - obtained from its URL.
-const SPREADSHEET_ID = '1feRnLYA4KYoe6NLtg50w3e33rYZpCJtS5cYhIo-YH38';
+const SPREADSHEET_ID = '1feRnLYA4KYoe6NLtg50w3e33rYZpCJtS5cYhIo-YH38'
 const SPREADSHEET_TITLE = 'Week-by-Week: Initiative Status'
-var data = 0
 // Todo: grab by the tracking URL instead
-var feature = 'MVP feature 2'
-var status_colors = {
+const STATUS_COLORS = {
```
Comment (mxstbr) [inline index.js]: Tiny consistency change: global constants should be uppercase. Just personal preference.

File: index.js [hunk 4]
```diff
@@ -55,26 +53,16 @@ async function accessSpreadsheet(status, url) {
       'max-col': sheet.colCount,
       'return-empty': true,
   })
+  
+  const index = cells.findIndex(cell => cell.value === url);
```
Comment (mxstbr) [inline index.js]: No need for a loop with a mutating variable, we can use the built-in `array.findIndex` method instead to find the index of the cell with the URL, which is much clearer and it avoids mutating any global variables. :tada:

File: index.js [hunk 5]
```diff
@@ -55,26 +53,16 @@ async function accessSpreadsheet(status, url) {
...
-async function updateStatusCell(cells, data, status){
-  do {
-    data++
-  }
-
-  // Get the last empty cell in the range
-  while (cells[data].value)
-  var cell = cells[data];
+async function updateStatusCell(cells, index, status) {
+  const emptyIndex = cells.findIndex(cell => !cell.value);
```
Comment (mxstbr) [inline index.js]: Again, no need for a `do/while` loop or anything, we can use the `findIndex` method again instead :100:

File: index.js (hunk 3 above)
Comment (dmleong) [reply index.js]: Ah, like a Ruby constant 👍

**Comment by dmleong** [comment]
@mxstbr THANK YOU!! Really appreciate you helping me get up to date on JavaScript! 💜

**Pull request merged**